cgt_calc/main.py

```diff
@@ -127,40 +136,149 @@ def add_acquisition(
+            spin_off,
+        )
+
+    def handle_spin_off(
+        self,
+        transaction: BrokerTransaction,
+    ) -> tuple[Decimal, Decimal, SpinOff]:
+        """
+        Handle spin off transaction.
+
+        Doc basing on SOLV spin off out of MMM.
+
+        # 1. Determine the Cost Basis (Acquisition Cost) of the SOLV Shares
+        In the UK, the cost basis (or acquisition cost) of the new SOLV shares
+        received from the spin-off needs to be determined. This is usually done
+        by apportioning part of the original cost basis of the MMM shares to
+        the new SOLV shares based on their market values at the time of the
+        spin-off.
+
+        ## Step-by-Step Allocation
+        * Find the Market Values:
+
+        Determine the market value of MMM shares and SOLV shares immediately
+        after the spin-off.
+
+        * Calculate the Apportionment:
+
+        Divide the market value of the MMM shares by the total market value of
+        both MMM and SOLV shares to find the percentage allocated to MMM.
+        Do the same for SOLV shares to find the percentage allocated to SOLV.
+
+        * Allocate the Original Cost Basis:
+
+        Multiply the original cost basis of your MMM shares by the respective
+        percentages to allocate the cost basis between the MMM and SOLV shares.
+
+        ## Example Allocation
+        * Original Investment:
+
+        Assume you bought 100 shares of MMM at £100 per share, so your total
+        cost basis is £10,000.
+
+        * Market Values Post Spin-off:
+
+        Assume the market value of MMM shares is £90 per share and SOLV shares
+        is £10 per share immediately after the spin-off.
+        The total market value is £90 + £10 = £100.
+
+        * Allocation Percentages:
+
+        Percentage allocated to MMM: 90/100 = 90%
+        Percentage allocated to SOLV: 10/100 = 10%
+
+        * Allocate Cost Basis:
+
+        Cost basis of MMM: £10,000 * 0.90 = £9,000
+        Cost basis of SOLV: £10,000 * 0.10 = £1,000
+
+        # 2. Determine the Holding Period
+        The holding period for the SOLV shares typically includes the holding
+        period of the original MMM shares. This means the date you acquired the
+        MMM shares will be used as the acquisition date for the SOLV shares.
+        """
+        symbol = transaction.symbol
+        quantity = transaction.quantity
+        assert symbol is not None
+        assert quantity is not None
+
+        price = self.price_fetcher.get_closing_price(symbol, transaction.date)
+        amount = quantity * price
+
+        while True:
+            # This would ideally be fetched from some stock DB but yfinance does not
+            # provide any info on SpinOffs
+            ticker = input(
+                "For a spin off, please enter the original ticker from which the new "
+                "stock was spinned off: "
+            )
+            if ticker in self.portfolio:
```

Comment:
Maybe show the list of options as well?

Reply:
That may be a long list, I guess we can show a trimmed one.